coderd/parameters.go

```diff
@@ -188,8 +168,195 @@ func (api *API) templateVersionDynamicParameters(rw http.ResponseWriter, r *http
 	}
 }
 
-func (api *API) getWorkspaceOwnerData(
+type previewFunction func(ctx context.Context, values map[string]string) (*preview.Output, hcl.Diagnostics)
+
+func prepareDynamicPreview(ctx context.Context, rw http.ResponseWriter, db database.Store, fc *files.Cache, tf database.TemplateVersionTerraformValue, templateVersion database.TemplateVersion, user database.User) (render previewFunction, closer func(), success bool) {
```

Comment:
This code was just extracted to this function. It was how dynamic params was originally written